### ISSUE TITLE
Collected FitSNAP small fixes and updates, Aug. 2024

### ISSUE THREAD

**Issue by megmcca** [FitSNAP/FitSNAP#258]
Hi all, 
I'm starting this issues thread to keep track of a few relatively small, but important fixes to keep in mind over the next few months.
The primary trigger for this are the fFtSNAP ParallelTools/shared memory issues we've been having on some of the newest Sandia clusters, but we can use this thread to catch up on a few other things too. 
I may move this to "Discussions" (or somewhere else) if folks think that's a better place.

If you have something to add to this list, put it as a comment or write to me and I'll update this description. @rohskopf, your input is very welcome and let us know if there's anything related you had planned that we can continue implementing.

Here they are in very rough order of importance (descending):

- Update FitSNAP output for logging memory usage, A matrix shape etc. and test on multiple machines
- Update FitSNAP install instructions (especially new LAMMPS/PACE instructions)
- Continue building out training set repo, split out if needed
- Implement a few reliable frozen dependency lists (e.g. Numpy 1.26, Pytorch ...)
- Consolidate conversion scripts (e.g. JSON to XYZ, JSON to ACE)
- Consider/discuss adding a few simpler build and install options for HPC (e.g. git submodule scripts, wheels)
- (add your suggestion here)

**Comment by rohskopf** [comment]
Hi @megmcca 

> Update FitSNAP install instructions (especially new LAMMPS/PACE instructions)

This is the biggest lowest hanging fruit. Has anyone used new LAMMPS/PACE for ACE fits?

**Comment by megmcca** [comment]
yep @rohskopf on the Sandia machines. the fits are working well there, at least. 

we also need to update a few of the (C)make flags/options that are now integrated into builds like LAMMPS_EXCEPTIONS and possibly some Python settings

we can make that a priority and address issue #256

**Comment by alphataubio** [comment]
>The primary trigger for this are the fFtSNAP ParallelTools/shared memory issues we've been having on some of the newest Sandia clusters

@megmcca i ran into a bug shared array not syncing. i traced it to (maybe) this commit https://github.com/FitSNAP/FitSNAP/commit/9119c4b1bea91eb1ce9fdaf1894e3855e8244ad4 from @rohskopf 

```
        #win = MPI.Win.Allocate_shared(self._nbytes, item_size, Intracomm_comm=self._comms[1][0])
        self.win = self.MPI.Win.Allocate_shared(self._nbytes, item_size, comm=self._comms[1][0])
```

```
*** [rank 0] create_shared_array(x_arrays)
*** comms [[<mpi4py.MPI.Intracomm object at 0x11707cd50>, 0, 3], [<mpi4py.MPI.Intracomm object at 0x1167de970>, 0, 3], [<mpi4py.MPI.Intracomm object at 0x11704a8b0>, 0, 1]]

*** [rank 1] create_shared_array(x_arrays)
*** comms [[<mpi4py.MPI.Intracomm object at 0x10aa00990>, 1, 3], [<mpi4py.MPI.Intracomm object at 0x10a9625e0>, 1, 3], [<mpi4py.MPI.Intracomm object at 0x10a9170c0>, 0, 1]]

*** [rank 2] create_shared_array(x_arrays)
*** comms [[<mpi4py.MPI.Intracomm object at 0x10ce70ab0>, 2, 3], [<mpi4py.MPI.Intracomm object at 0x10cdd2af0>, 2, 3], [<mpi4py.MPI.Intracomm object at 0x10cdd2760>, 0, 1]]
```

i dont have a fix yet
